File: Creational/Abstract Factory/JavaScript/js/suvFactory.js
import IAutoBodyFactory from './factory.js';
import { SUVLeftDoor, SUVRightDoor, SUVHood } from './product.js';

// Concrete Factory for SUV
class SUVFactory extends IAutoBodyFactory {
    createLeftDoor() {
        return new SUVLeftDoor();
    }

    createRightDoor() {
        return new SUVRightDoor();
    }

    createHood() {
        return new SUVHood();
    }
}

// Exporting the class as a default export
export default SUVFactory;

File: Creational/Abstract Factory/C#/SUVFactory.cs
namespace AutoBodyFactory {
    public class SUVFactory : IAutoBodyFactory {
        public IAutoBodyPart CreateLeftDoor() {
            return new SUVLeftDoor();
        }

        public IAutoBodyPart CreateRightDoor() {
            return new SUVRightDoor();
        }

        public IAutoBodyPart CreateHood() {
            return new SUVHood();
        }
    }
}
